src/internal_modules/roc_core/string_list.cpp

```diff
@@ -24,15 +25,19 @@ size_t StringList::size() const {
 
 const char* StringList::front() const {
     if (size_) {
-        return &data_[0];
+        const Header* first_header =
+            ROC_CONTAINER_OF(const_cast<char*>(&data_[0]), Header, len);
+        return first_header->str_;
     } else {
         return NULL;
     }
 }
```

Comment:
We can make thing more clean if we'll maintain two class fields:

```cpp
    Header* front_;
    Header* back_;
```

Then front(), for example, will look as:

```cpp
const char* StringList::front() const {
    if (front_) {
        return front_->str;
    } else {
        return NULL;
    }
}
```